File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    for p in participant:
        if p not in dic:
            dic[p] = 1
        else:
            dic[p] +=1
    for c in completion:
        dic[c] -=1
    for i in dic:
        if dic[i] != 0 :
            answer = i
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String,Integer> map = new HashMap<>();
        for (int i=0;i<participant.length;i++) {
            if (map.containsKey(participant[i])) {
                map.put(participant[i],map.get(participant[i]) + 1);
            }else {
                map.put(participant[i],1);
            }
        }
        for (int i=0;i<completion.length;i++) {
            map.put(completion[i],map.get(completion[i])-1);
        }
        for (String s: map.keySet()) {
            if (map.get(s) == 1) {
                answer = s;
            }
        }
        return answer;
    }
}